core/git-access.c

```diff
@@ -664,6 +665,22 @@ static git_repository *update_local_repo(const char *localdir, const char *remot
 			report_error("Unable to open git cache repository at %s: %s", localdir, giterr_last()->message);
 		return NULL;
 	}
+
+	/* Check the HEAD being the right branch */
+	if (!git_repository_head(&head, repo)) {
+		const char *name;
+		if (!git_branch_name(&name, head)) {
+			if (strcmp(name, branch)) {
+				char *branchref = format_string("refs/heads/%s", branch);
+
+				report_error("Setting cache branch from '%s' to '%s'", name, branch);
```

Comment:
I wonder if this is something where we should show a RED ERROR to the user, or if we shouldn't just silently fix it for them... I mean, what would they do with the knowledge that the cache branch changed?
Imagine Andy and Wendie seeing that message... :-)